controlplane/src/core/controllers/auth.ts

```diff
@@ -158,6 +164,46 @@ const plugin: FastifyPluginCallback<AuthControllerOptions> = function Auth(fasti
       return insertedSessions[0];
     });
 
+    const orgs = await opts.organizationRepository.memberships({
+      userId,
+    });
+    if (orgs.length === 0) {
+      await opts.keycloakClient.authenticateClient();
+
+      const organizationSlug = uid(8);
+
+      await opts.keycloakClient.seedGroup({ userID: userId, organizationSlug, realm: opts.keycloakRealm });
+
+      await opts.db.transaction(async (db) => {
+        const insertedOrg = await db
```

Comment:
You can create the organization repository here. Just instantiate a repo with a custom db context.

Reply:
Done